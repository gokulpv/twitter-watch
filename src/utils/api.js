const BACKEND_BASE_URL = "http://localhost:3000";

export const get = async (pathname) => {
  const response = await fetch(`${BACKEND_BASE_URL}/${pathname}`);
  const data = await response.json();
  if (!response.ok) {
    // Can be made to throw specific error messages
    throw new Error("Some error occured.Please check the url");
  }
  return data;
};
