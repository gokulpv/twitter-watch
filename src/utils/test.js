export const findByTestId = (component, id) => {
  const wrapper = component.find(`[data-testid='${id}']`);
  return wrapper;
};
