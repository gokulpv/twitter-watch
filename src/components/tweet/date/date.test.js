import React from "react";
import Date from "./Date";
import { shallow } from "enzyme";
import { findByTestId } from "../../../utils/test";

describe("Date component", () => {
  let component;
  // Setup the component for test
  beforeEach(() => {
    const expectedProps = { dateString: "Tue Sep 28 20:29:31 +0000 2021" };
    component = shallow(<Date {...expectedProps} />);
  });

  it("should render Date component", () => {
    expect(component).toMatchSnapshot();
  });

  it("identify a date string inside", () => {
    expect(findByTestId(component, "date-string").text()).not.toBe("");
  });
});
