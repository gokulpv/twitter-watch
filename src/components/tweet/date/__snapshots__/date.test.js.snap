// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date component should render Date component 1`] = `ShallowWrapper {}`;
